### PR TITLE
EVG-15110: increased checks and logging to prevent collision of host drawdown jobs

### PR DESCRIPTION
**Pull request by hhoke**
[EVG-15110](https://jira.mongodb.org/browse/EVG-15110)

### Description 
1) We appeared to be having a bunch of collisions due to the fact that drawdown jobs are named only after the start time, resulting in drastically reduced effectiveness system-wide
2) Drawdown jobs were being run for static distros, which should never be the case. I had assumed the host allocator only ran for spawnable distros, but there is now a check built in to make sure we only create host drawdown jobs for spawnable distros.
3) A huge portion of drawdown jobs end up killing 0 hosts. I added logging to help figure out why.

### Testing 
  none

### CODE REVIEW

**Comment by ybrill** [comment]
We definitely run the host allocator for static distros too. I think the reason we don't try to spawn them is [the return here](https://github.com/evergreen-ci/evergreen/blob/db3f306dac84b094a7a55942a0c6f03995a916a3/scheduler/utilization_based_host_allocator.go#L151-L153).

File: units/host_allocator.go [hunk 1]
```diff
@@ -305,7 +306,7 @@ func (j *hostAllocatorJob) setTargetAndTerminate(ctx context.Context, numUpHosts
 			DistroID:     distro.Id,
 			NewCapTarget: newCapTarget,
 		}
-		err := amboy.EnqueueUniqueJob(ctx, j.env.RemoteQueue(), NewHostDrawdownJob(j.env, drawdownInfo, utility.RoundPartOfMinute(1).Format(TSFormat)))
+		err := amboy.EnqueueUniqueJob(ctx, j.env.RemoteQueue(), NewHostDrawdownJob(j.env, drawdownInfo, utility.RoundPartOfMinute(1).Format(TSFormat)+distro.Id))
```
Comment (ybrill) [inline units/host_allocator.go:309]: an alternative to doing it this way is to encorporate the distro id in the id and scope [here](https://github.com/evergreen-ci/evergreen/blob/db3f306dac84b094a7a55942a0c6f03995a916a3/units/host_drawdown.go#L67-L68)

**Review by ybrill — COMMENTED** [review]
perhaps we should also log when `checkTerminationExemptions` returns true?

**Comment by hhoke** [comment]
> perhaps we should also log when `checkTerminationExemptions` returns true?

I can't think of a reason to do this -- we already log or have a descriptive error for most of the cases where checkTerminationExemptions returns true, and j.Terminated keeps a reliable record of how many hosts have ben terminated.

File: units/host_allocator.go (hunk 1 above)
Comment (ybrill) [reply units/host_allocator.go:309]: This definitely _works_, but I think it's more typical to assemble the id in the job's "constructor". Doing it there has the advantage that the id is consistent no matter where the job is created. 
A couple of examples [here](https://github.com/evergreen-ci/evergreen/blob/ea2993ee93fc6fc77fd6dbc6ef33063f74d6ec4f/units/host_allocator.go#L62), [here](https://github.com/evergreen-ci/evergreen/blob/ea2993ee93fc6fc77fd6dbc6ef33063f74d6ec4f/units/scheduler.go#L50), [here](https://github.com/evergreen-ci/evergreen/blob/ea2993ee93fc6fc77fd6dbc6ef33063f74d6ec4f/units/provisioning_user_data_done.go#L64), etc.

Comment (hhoke) [reply units/host_allocator.go]: I changed it to fit this format

**Review by ybrill — APPROVED** [review]

**Pull request merged**